app/src/main/java/com/radixdlt/android/ui/activity/WebViewActivity.kt

```diff
@@ -0,0 +1,54 @@
+package com.radixdlt.android.ui.activity
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.MenuItem
+import android.webkit.WebViewClient
+import androidx.appcompat.widget.Toolbar
+import com.radixdlt.android.R
+import kotlinx.android.synthetic.main.activity_webview.*
+import kotlinx.android.synthetic.main.barcode_capture.*
+
+/**
+ * This Activity is used as a fallback when there is no browser installed that supports
+ * Chrome Custom Tabs
+ */
+class WebViewActivity : BaseActivity() {
+
+    @SuppressLint("SetJavaScriptEnabled")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_webview)
+
+        val url = intent.getStringExtra(EXTRA_URL)
+        setToolBar(url)
+
+        webview.webViewClient = WebViewClient()
+        val webSettings = webview.settings
+        webSettings.javaScriptEnabled = true
+
+        webview.loadUrl(url)
+    }
+
+    private fun setToolBar(url: String?) {
+        setSupportActionBar(toolbar as Toolbar)
+        supportActionBar!!.setHomeButtonEnabled(true)
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+        supportActionBar!!.title = url
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            // Respond to the action bar's Up/Home button
+            android.R.id.home -> {
+                finish()
```

Comment:
Is there any browsing history to honor here before we `finish()`?

Reply:
Not sure, this is the fallback in case the device hasn't got any chrome compatible browser or something like that which will be a very small minority nowadays.

Not sure about the browsing history, I don't think the webview stores anything like that as it's function is to simply display a website and loads fresh every time as far as I know.

Reply:
Yes, the `WebView` does handle history, which is hinted about here:
https://developer.android.com/reference/android/webkit/WebView.html#goBack()

This isn't an issue if we have absolute and total control of the web resource being displayed (no external links, etc). But as soon as there is a single link to an external resource, there is a history building up. The link issue is also another example of things that make it almost impossible to disguise web UI as native UI and a good reason why one shouldn't even try.

Reply:
Will check this out and keep it in mind for the future!

Reply:
I had to look into it and think this solves the potential problem!